app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java

```diff
@@ -240,51 +219,61 @@ private void createDateView(){
         }
     }
 
-    private void createSportView(){
+    private void createSportView() {
+        // Type of sport of the activity
         TextView sportView = (TextView) findViewById(R.id.activity_sport_description);
-        if(act != null){
+        if (act != null) {
             sportView.setText(act.getSport().toString());
         }
     }
 
-    private void createDurationView(){
+    private void createDurationView() {
+        // Duration of the activity
         TextView durationView = (TextView) findViewById(R.id.activity_duration_description);
-        if(act != null){
-            durationView.setText(String.valueOf((int) act.getDuration()));
+        if (act != null) {
+            durationView.setText(String.valueOf((double) Math.round(act.getDuration() * 100) / 100));
         }
     }
 
-    private void createOrganizerView(){
+    private void createOrganizerView() {
+        // Organizer of the activity
         organizerView = (TextView) findViewById(R.id.activity_organisator_description);
-        if(act != null){
+        if (act != null) {
             organizerView.setText(act.getOrganizerId());
         }
     }
 
-    private void createAddressView(){
-        // address from the activity
+    private void createAddressView() {
+        // Address of the activity
         TextView addressView = (TextView) findViewById(R.id.activity_address_description);
         if (act != null) {
             addressView.setText(act.getAddress());
         }
     }
 
-    public void onClick(View v) {
```

Comment:
Why was `onClick()` removed?

Reply:
Nevermind, I saw the changes in the XML

Reply:
Yes, the reason is that this was not an appropriate name for the functionality to register to the sport activity. "registerToActivity" is more straight forward!